test/config.test.js

```diff
@@ -270,6 +270,57 @@ describe('javascript config', () => {
 
 });
 
+
+describe('`logger` config', () => {
+    // Arrange
+    const dir = 'A directory';
+    const subDir = 'A sub directory';
+    const file = 'A file';
+
+    beforeEach(() => {
```

Comment:
I personally find this style harder to follow than the existing tests. When everything is declared within a single file you don't have to go scanning back up to see if there is a `beforeEach` block. It doesn't add much value that I can see?

At least before the tests act more like documentation; someone can find the property they are interested in and see exactly how the properties are updated at a glance

Reply:
Also, are we not waiting until we agree on a better to start rewriting the tests? Seems like we're introducing inconsistency because we disagree on the style. Wouldn't it be better to get on the same page before updating the tests?

Reply:
Hi @DamianMullins I totally agree about not rewriting the tests until we agree on a structure which is why I have reverted them, but the reason I've added the `beforeEach` is because my tests fail on the `.toBeDefined` scenario if the `config.update` isn't updated before the second and the remaining `IT` block statements. I have removed the nested describes as we agreed but without repeating and updating the config multiple times for each `IT` and to keep things `DRY` so as to not repeat code multiple times to make it more readable. If you would prefer me to add them to each `IT` instead happy to do so?

Reply:
OK, so it sounds like you have another issue in the tests, I'll have a quick look this afternoon so I fully understand the issue. 

As for the comments about DRY, I'm always on the fence with applying the principle in unit tests as it tends to lead towards code which harder to reason about. Let's arrange something soon so we can discuss and put some examples together?

Reply:
@simonsandersje Sounds like we need that doodle to sort out the next UI Guild 😉

Reply:
@kevinrodrigues the issue you're having with the config value on the second run is because you need to [add a line to the `update()` function](https://github.com/justeat/gulp-build-fozzie/blob/d73376492762e79ff26c77a5381dda1e513f1dfb/config.js#L180) which allows the property to update correctly.

```js
logger: Object.assign(config.logger, options.logger)
```

Reply:
@DamianMullins if we use the update config way then we don't need to mock the `Arrange` and `Act` setup or invoke the `config.update` again because you can read directly from the config object inside the `IT` statements. Is this ok?

```
describe('`logger` config', () => {
    it('should exist', () => {
        // Assert
        expect(config.logger).toBeDefined();
    });

    // `logger.dir`
    it('`dir` should exist', () => {
        // Assert
        expect(config.logger.dir).toBeDefined();
    });

    it('`dir` can be updated', () => {
        // Assert
        expect(config.logger.dir).toBe(config.logger.dir);
    });

    // `logger.subDir`
    it('`subDir` should exist', () => {
        // Assert
        expect(config.logger.subDir).toBeDefined();
    });

    it('`subDir` can be updated', () => {
        // Assert
        expect(config.logger.subDir).toBe(config.logger.subDir);
    });

    // `logger.file`
    it('`file` should exist', () => {
        // Assert
        expect(config.logger.file).toBeDefined();
    });

    it('`file` can be updated', () => {
        // Assert
        expect(config.logger.file).toBe(config.logger.file);
    });
});
```

Reply:
Where are you updating the value in the code above? In the example I posted, the tests explicitly update the original value.

Reply:
Why not keep it the same as everything else in the test file? I don't understand the need to make it different at the moment?

Reply:
Done.